chainntnfs/txconfnotifier.go

```diff
@@ -335,7 +551,14 @@ func (tcn *TxConfNotifier) ConnectTip(blockHash *chainhash.Hash,
 		txsToUpdateHints = append(txsToUpdateHints, confirmedTx)
 	}
 out:
-	for maybeUnconfirmedTx := range tcn.confNotifications {
+	for maybeUnconfirmedTx, confSet := range tcn.confNotifications {
```

Comment:
This logic is a bit hard to follow currently. Would it be equivalent to do
```golang
var txsToUpdateHints []chainhash.Hash
for maybeUnconfirmedTx, confSet := range tcn.confNotifications {
                if confSet.rescanStatus != rescanComplete {
			continue
		}
                txsToUpdateHints = append(txsToUpdateHints, maybeUnconfirmedTx)
}
```
?

Reply:
fixed wilmer's branch https://github.com/lightningnetwork/lnd/pull/2004/commits/e47bbaff3a6d1fc0f39f361c01557fa986d3653f